### PR TITLE
[Swift] Day 16 Completed.

### PR DESCRIPTION
**Pull request by swinful**

### CODE REVIEW

File: SWIFT/day16.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    for (var i = 1; i < elements.count; ++i) {
+      let difference = abs(elements[i-1] - elements[i])
+      if difference == minimumDifference {
+        minimumPairs.append(elements[i-1])
+        minimumPairs.append(elements[i])
+      }
+
+      if difference < minimumDifference {
+        minimumDifference = difference
+        minimumPairs      = [Int]()
```
Comment (Ashok-Varma) [inline SWIFT/day16.swift]: minimumPairs.removeAll()

File: SWIFT/day16.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import Foundation
+
+// Class Difference
+class Difference {
+
+  // MARK: Private Instance variables
+  private var elements          = [Int]()
+  private var minimumPairs      = [Int]()
+  private var minimumDifference = 0
```
Comment (Ashok-Varma) [inline SWIFT/day16.swift]: private var minimumDifference = Int.max

initial diff should be Int.max not 0

File: SWIFT/day16.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    minimumPairs.forEach { element in
+      print("\(element)", terminator: " ")
+    }
+  }
+}
+
+// MARK: Main
+let n = Int(readLine()!)!
+let a = readLine()!.characters.split(" ").map {Int(String($0))!}.sort()
+
```
Comment (Ashok-Varma) [inline SWIFT/day16.swift:44]: Difference is not a failable init. so conditional binding will fail.

code should be like this

let d = Difference(a: a)
d.computeMinimumDifference()
d.printPairs()

Comment (swinful) [inline SWIFT/day16.swift:44]: Difference was a failable init. I forgot to remove the conditional binding when I removed the failable init. Will correct this.

File: SWIFT/day16.swift (hunk 2 above)
Comment (swinful) [inline SWIFT/day16.swift]: Is there a test case that will cause the wrong output if initial diff is set to 0?

File: SWIFT/day16.swift (hunk 1 above)
Comment (swinful) [inline SWIFT/day16.swift]: Curious here which is optimal or if they are the same cost. Actually, I did not need to re-initialize it with (). Simply doing [] would have been more effective. The empty brackets, [], is used in The Swift Programming guide. See Collection Types -> Arrays.

File: SWIFT/day16.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        minimumPairs      = []
+        minimumPairs.append(elements[i-1])
+        minimumPairs.append(elements[i])
+      }
+    }
+  }
+
+  // Print out the pairs
+  func printPairs() {
+    minimumPairs.forEach { element in
```
Comment (della-90) [inline SWIFT/day16.swift]: If you want you can shorten the closure like this: `minimumPairs.forEach { print("\($0)", terminator: " ") }`

Comment (swinful) [inline SWIFT/day16.swift]: Thanks. That works as well.

**Pull request merged**